#### Name: Anusha Aligeti
#### Course: BUDT704
#### Section: 0503
#### Date: 11/3/2023

In [1]:
import pandas as pd #import pandas library
import numpy as np #import numpy library

#### 1. Loading the Data From the Excel
- By `pd.read_excel`, we read the excel and assigned it to a dataframe called df

In [2]:
df = pd.read_excel('HW5_Socially Responsible ETFs.xlsx') #load the excel file
df

Ticker                                          Fund Name  \
0     ESGU                     iShares ESG Aware MSCI USA ETF   
1     ESGD                    iShares ESG Aware MSCI EAFE ETF   
2     ESGE                      iShares ESG Aware MSCI EM ETF   
3     ICLN                    iShares Global Clean Energy ETF   
4     ESGV                        Vanguard ESG U.S. Stock ETF   
..     ...                                                ...   
161   EEMX  SPDR MSCI Emerging Markets Fossil Fuel Reserve...   
162   EFAX       SPDR MSCI EAFE Fossil Fuel Reserves Free ETF   
163    SHE               SPDR SSGA Gender Diversity Index ETF   
164   SPYX         SPDR S&P 500 Fossil Fuel Reserves Free ETF   
165   LOWC               SPDR MSCI ACWI Low Carbon Target ETF   

                           Issuer      AUM  Expense Ratio 3-Mo TR  \
0                       Blackrock  $23.43B         0.0015  0.0348   
1                       Blackrock   $7.11B         0.0020  0.0165   
2                       Blackrock   $7.00B         0.0025  0.0147   
3                       Blackrock   $6.46B         0.0042  0.0612   
4                        Vanguard   $5.72B         0.0012  0.0355   
..                            ...      ...            ...     ...   
161  State Street Global Advisors       --         0.0030  0.0108   
162  State Street Global Advisors       --         0.0020   0.011   
163  State Street Global Advisors       --         0.0020   0.029   
164  State Street Global Advisors       --         0.0020  0.0337   
165  State Street Global Advisors       --         0.0020  0.0288   

                                               Segment  
0                          Equity: U.S. - Total Market  
1    Equity: Developed Markets Ex-North America - T...  
2              Equity: Emerging Markets - Total Market  
3                      Equity: Global Renewable Energy  
4                          Equity: U.S. - Total Market  
..                                                 ...  
161            Equity: Emerging Markets - Total Market  
162  Equity: Developed Markets Ex-North America - T...  
163                           Equity: U.S. - Large Cap  
164                           Equity: U.S. - Large Cap  
165                          Equity: Global Low Carbon  

[166 rows x 7 columns]

#### 1b) Replacing the missing values:
- Since in the excel sheet, all missing values are shown with a dash `--` we replace the dashes with NaN values using numpy library

In [3]:
df = df.replace('--', np.nan) #replacing missing values with NaN

#### 1c) Displaying the first 20 funds including all the known data:
- Using `.head()` to display the first rows of the dataframe, Since all the columns in the dataframe to be printed, we use `df.head(20)`

In [4]:
df.head(20) #printing the first 20 funds

Ticker                                          Fund Name  \
0    ESGU                     iShares ESG Aware MSCI USA ETF   
1    ESGD                    iShares ESG Aware MSCI EAFE ETF   
2    ESGE                      iShares ESG Aware MSCI EM ETF   
3    ICLN                    iShares Global Clean Energy ETF   
4    ESGV                        Vanguard ESG U.S. Stock ETF   
5    XSOE  WisdomTree Emerging Markets ex-State-Owned Ent...   
6    SUSL                   iShares ESG MSCI USA Leaders ETF   
7    SUSA                    iShares MSCI USA ESG Select ETF   
8    USSG       Xtrackers MSCI U.S.A. ESG Leaders Equity ETF   
9     DSI                    iShares MSCI KLD 400 Social ETF   
10    TAN                                  Invesco Solar ETF   
11   QCLN  First Trust NASDAQ Clean Edge Green Energy Ind...   
12   VSGX               Vanguard ESG International Stock ETF   
13    PBW                Invesco WilderHill Clean Energy ETF   
14   EAGG          iShares ESG Aware U.S. Aggregate Bond ETF   
15   ESML           iShares ESG Aware MSCI USA Small-Cap ETF   
16   NULV                     Nuveen ESG Large-Cap Value ETF   
17   CRBN            iShares MSCI ACWI Low Carbon Target ETF   
18   NUSC                           Nuveen ESG Small-Cap ETF   
19   KRBN                      KraneShares Global Carbon ETF   

               Issuer      AUM  Expense Ratio  3-Mo TR  \
0           Blackrock  $23.43B         0.0015   0.0348   
1           Blackrock   $7.11B         0.0020   0.0165   
2           Blackrock   $7.00B         0.0025   0.0147   
3           Blackrock   $6.46B         0.0042   0.0612   
4            Vanguard   $5.72B         0.0012   0.0355   
5          WisdomTree   $4.28B         0.0032  -0.0013   
6           Blackrock   $4.18B         0.0010   0.0527   
7           Blackrock   $4.06B         0.0025   0.0439   
8       Deutsche Bank   $4.00B         0.0010   0.0498   
9           Blackrock   $3.75B         0.0025   0.0540   
10            Invesco   $3.55B         0.0069   0.0902   
11        First Trust   $2.83B         0.0060   0.1126   
12           Vanguard   $2.82B         0.0017   0.0127   
13            Invesco   $1.89B         0.0061   0.0300   
14          Blackrock   $1.63B         0.0010  -0.0110   
15          Blackrock   $1.21B         0.0017   0.0437   
16  Nuveen Securities   $1.18B         0.0025   0.0343   
17          Blackrock   $1.17B         0.0020   0.0282   
18  Nuveen Securities   $1.11B         0.0030   0.0356   
19               CICC   $1.06B         0.0079   0.1732   

                                              Segment  
0                         Equity: U.S. - Total Market  
1   Equity: Developed Markets Ex-North America - T...  
2             Equity: Emerging Markets - Total Market  
3                     Equity: Global Renewable Energy  
4                         Equity: U.S. - Total Market  
5             Equity: Emerging Markets - Total Market  
6                         Equity: U.S. - Total Market  
7                         Equity: U.S. - Total Market  
8                         Equity: U.S. - Total Market  
9                         Equity: U.S. - Total Market  
10                    Equity: Global Renewable Energy  
11                      Equity: U.S. Renewable Energy  
12              Equity: Global Ex-U.S. - Total Market  
13                    Equity: Global Renewable Energy  
14  Fixed Income: U.S. - Broad Market, Broad-based...  
15                           Equity: U.S. - Small Cap  
16                     Equity: U.S. - Large Cap Value  
17                          Equity: Global Low Carbon  
18                           Equity: U.S. - Small Cap  
19                 Commodities: Energy Carbon Credits

### Processing

#### 2) AUM cleaning:
- Since all the values of the AUM variable don't have the same units, we change all to one unit i.e to millions
- Some of the values in the column are NaN and they don't have to be changed
- We create a function called `billion_to_million` to convert all the billion values to million
- The million and NaN values remain the same
- To find out the billion values we use regular expressions to create a pattern and find billion values and convert them to millions

In [5]:
import re #import regular expressions library
df['AUM'] = df['AUM'].astype(str).str.replace('[\$]', '', regex=True) #replace dollar sign with ''
df['AUM']

0      23.43B
1       7.11B
2       7.00B
3       6.46B
4       5.72B
        ...  
161       nan
162       nan
163       nan
164       nan
165       nan
Name: AUM, Length: 166, dtype: object

In [6]:
#defining a function which turns billions to millions

def billion_to_million(s): 
    if s == np.NaN:         #if value is NaN keep it same
        result = np.NaN
    else:
        pattern = r'(\d+\.\d+)B'   #regular expression pattern for billion
        result = re.sub(pattern, lambda match: str(float(match.group(1)) * 1000),str(s))
    return result

- Applying the function onto the `AUM` column of the dataframe to convert the billion values

In [7]:
df['AUM'] = df['AUM'].apply(billion_to_million)  #appying the function to AUM

- As the AUM column contains all millions now, but there are units from the original million values, strip the million sign from the dataframe and replace M with ''
- For the Further analysis, we may have to work with AUM data, so we convert the column to numeric so we can perform operations on it

In [8]:
df['AUM'] = df['AUM'].astype(str).str.replace('[M]', '', regex=True) #replacing M with ''
df['AUM'] = pd.to_numeric(df['AUM'], errors = 'coerce') #converting the AUM column into numeric data

- By applying the function onto the column 'AUM', all the values of the column had to be converted to datatype `str`
- The NaN values also turn into strings so we replace them back to np.NaN

In [9]:
df = df.replace('nan', np.nan)   #replacing nan to NaN
df

Ticker                                          Fund Name  \
0     ESGU                     iShares ESG Aware MSCI USA ETF   
1     ESGD                    iShares ESG Aware MSCI EAFE ETF   
2     ESGE                      iShares ESG Aware MSCI EM ETF   
3     ICLN                    iShares Global Clean Energy ETF   
4     ESGV                        Vanguard ESG U.S. Stock ETF   
..     ...                                                ...   
161   EEMX  SPDR MSCI Emerging Markets Fossil Fuel Reserve...   
162   EFAX       SPDR MSCI EAFE Fossil Fuel Reserves Free ETF   
163    SHE               SPDR SSGA Gender Diversity Index ETF   
164   SPYX         SPDR S&P 500 Fossil Fuel Reserves Free ETF   
165   LOWC               SPDR MSCI ACWI Low Carbon Target ETF   

                           Issuer      AUM  Expense Ratio  3-Mo TR  \
0                       Blackrock  23430.0         0.0015   0.0348   
1                       Blackrock   7110.0         0.0020   0.0165   
2                       Blackrock   7000.0         0.0025   0.0147   
3                       Blackrock   6460.0         0.0042   0.0612   
4                        Vanguard   5720.0         0.0012   0.0355   
..                            ...      ...            ...      ...   
161  State Street Global Advisors      NaN         0.0030   0.0108   
162  State Street Global Advisors      NaN         0.0020   0.0110   
163  State Street Global Advisors      NaN         0.0020   0.0290   
164  State Street Global Advisors      NaN         0.0020   0.0337   
165  State Street Global Advisors      NaN         0.0020   0.0288   

                                               Segment  
0                          Equity: U.S. - Total Market  
1    Equity: Developed Markets Ex-North America - T...  
2              Equity: Emerging Markets - Total Market  
3                      Equity: Global Renewable Energy  
4                          Equity: U.S. - Total Market  
..                                                 ...  
161            Equity: Emerging Markets - Total Market  
162  Equity: Developed Markets Ex-North America - T...  
163                           Equity: U.S. - Large Cap  
164                           Equity: U.S. - Large Cap  
165                          Equity: Global Low Carbon  

[166 rows x 7 columns]

- Displaying the segment column to study how the segment column can be split into different columns

In [10]:
df['Segment']

0                            Equity: U.S. - Total Market
1      Equity: Developed Markets Ex-North America - T...
2                Equity: Emerging Markets - Total Market
3                        Equity: Global Renewable Energy
4                            Equity: U.S. - Total Market
                             ...                        
161              Equity: Emerging Markets - Total Market
162    Equity: Developed Markets Ex-North America - T...
163                             Equity: U.S. - Large Cap
164                             Equity: U.S. - Large Cap
165                            Equity: Global Low Carbon
Name: Segment, Length: 166, dtype: object

#### 3) Splitting the Sement Column
- By studying the segment column, we notice that `:` seperates the asset class with the other market and segment
- We split the Asset class by `:` and place it into a new column and the rest of the split data to a segment column

In [11]:
#splitting segment into segment and asset class
df[['Asset Class','ETF Segment']] = df['Segment'].str.split(':', expand=True)

In [12]:
#checking the dataframe
df.head(20)

Ticker                                          Fund Name  \
0    ESGU                     iShares ESG Aware MSCI USA ETF   
1    ESGD                    iShares ESG Aware MSCI EAFE ETF   
2    ESGE                      iShares ESG Aware MSCI EM ETF   
3    ICLN                    iShares Global Clean Energy ETF   
4    ESGV                        Vanguard ESG U.S. Stock ETF   
5    XSOE  WisdomTree Emerging Markets ex-State-Owned Ent...   
6    SUSL                   iShares ESG MSCI USA Leaders ETF   
7    SUSA                    iShares MSCI USA ESG Select ETF   
8    USSG       Xtrackers MSCI U.S.A. ESG Leaders Equity ETF   
9     DSI                    iShares MSCI KLD 400 Social ETF   
10    TAN                                  Invesco Solar ETF   
11   QCLN  First Trust NASDAQ Clean Edge Green Energy Ind...   
12   VSGX               Vanguard ESG International Stock ETF   
13    PBW                Invesco WilderHill Clean Energy ETF   
14   EAGG          iShares ESG Aware U.S. Aggregate Bond ETF   
15   ESML           iShares ESG Aware MSCI USA Small-Cap ETF   
16   NULV                     Nuveen ESG Large-Cap Value ETF   
17   CRBN            iShares MSCI ACWI Low Carbon Target ETF   
18   NUSC                           Nuveen ESG Small-Cap ETF   
19   KRBN                      KraneShares Global Carbon ETF   

               Issuer      AUM  Expense Ratio  3-Mo TR  \
0           Blackrock  23430.0         0.0015   0.0348   
1           Blackrock   7110.0         0.0020   0.0165   
2           Blackrock   7000.0         0.0025   0.0147   
3           Blackrock   6460.0         0.0042   0.0612   
4            Vanguard   5720.0         0.0012   0.0355   
5          WisdomTree   4280.0         0.0032  -0.0013   
6           Blackrock   4180.0         0.0010   0.0527   
7           Blackrock   4060.0         0.0025   0.0439   
8       Deutsche Bank   4000.0         0.0010   0.0498   
9           Blackrock   3750.0         0.0025   0.0540   
10            Invesco   3550.0         0.0069   0.0902   
11        First Trust   2830.0         0.0060   0.1126   
12           Vanguard   2820.0         0.0017   0.0127   
13            Invesco   1890.0         0.0061   0.0300   
14          Blackrock   1630.0         0.0010  -0.0110   
15          Blackrock   1210.0         0.0017   0.0437   
16  Nuveen Securities   1180.0         0.0025   0.0343   
17          Blackrock   1170.0         0.0020   0.0282   
18  Nuveen Securities   1110.0         0.0030   0.0356   
19               CICC   1060.0         0.0079   0.1732   

                                              Segment   Asset Class  \
0                         Equity: U.S. - Total Market        Equity   
1   Equity: Developed Markets Ex-North America - T...        Equity   
2             Equity: Emerging Markets - Total Market        Equity   
3                     Equity: Global Renewable Energy        Equity   
4                         Equity: U.S. - Total Market        Equity   
5             Equity: Emerging Markets - Total Market        Equity   
6                         Equity: U.S. - Total Market        Equity   
7                         Equity: U.S. - Total Market        Equity   
8                         Equity: U.S. - Total Market        Equity   
9                         Equity: U.S. - Total Market        Equity   
10                    Equity: Global Renewable Energy        Equity   
11                      Equity: U.S. Renewable Energy        Equity   
12              Equity: Global Ex-U.S. - Total Market        Equity   
13                    Equity: Global Renewable Energy        Equity   
14  Fixed Income: U.S. - Broad Market, Broad-based...  Fixed Income   
15                           Equity: U.S. - Small Cap        Equity   
16                     Equity: U.S. - Large Cap Value        Equity   
17                          Equity: Global Low Carbon        Equity   
18                           Equity: U.S. - Small Cap        Equity   
19                 

- ` - `seperates the market with the segment, so we split the string and place it into a new column market

In [13]:
#splitting segment again to market and segment
df[['Market','ETF Segment']] = df['ETF Segment'].str.split(' - ', n=1, expand=True)

In [14]:
#checking the dataframe
df.head(20)

Ticker                                          Fund Name  \
0    ESGU                     iShares ESG Aware MSCI USA ETF   
1    ESGD                    iShares ESG Aware MSCI EAFE ETF   
2    ESGE                      iShares ESG Aware MSCI EM ETF   
3    ICLN                    iShares Global Clean Energy ETF   
4    ESGV                        Vanguard ESG U.S. Stock ETF   
5    XSOE  WisdomTree Emerging Markets ex-State-Owned Ent...   
6    SUSL                   iShares ESG MSCI USA Leaders ETF   
7    SUSA                    iShares MSCI USA ESG Select ETF   
8    USSG       Xtrackers MSCI U.S.A. ESG Leaders Equity ETF   
9     DSI                    iShares MSCI KLD 400 Social ETF   
10    TAN                                  Invesco Solar ETF   
11   QCLN  First Trust NASDAQ Clean Edge Green Energy Ind...   
12   VSGX               Vanguard ESG International Stock ETF   
13    PBW                Invesco WilderHill Clean Energy ETF   
14   EAGG          iShares ESG Aware U.S. Aggregate Bond ETF   
15   ESML           iShares ESG Aware MSCI USA Small-Cap ETF   
16   NULV                     Nuveen ESG Large-Cap Value ETF   
17   CRBN            iShares MSCI ACWI Low Carbon Target ETF   
18   NUSC                           Nuveen ESG Small-Cap ETF   
19   KRBN                      KraneShares Global Carbon ETF   

               Issuer      AUM  Expense Ratio  3-Mo TR  \
0           Blackrock  23430.0         0.0015   0.0348   
1           Blackrock   7110.0         0.0020   0.0165   
2           Blackrock   7000.0         0.0025   0.0147   
3           Blackrock   6460.0         0.0042   0.0612   
4            Vanguard   5720.0         0.0012   0.0355   
5          WisdomTree   4280.0         0.0032  -0.0013   
6           Blackrock   4180.0         0.0010   0.0527   
7           Blackrock   4060.0         0.0025   0.0439   
8       Deutsche Bank   4000.0         0.0010   0.0498   
9           Blackrock   3750.0         0.0025   0.0540   
10            Invesco   3550.0         0.0069   0.0902   
11        First Trust   2830.0         0.0060   0.1126   
12           Vanguard   2820.0         0.0017   0.0127   
13            Invesco   1890.0         0.0061   0.0300   
14          Blackrock   1630.0         0.0010  -0.0110   
15          Blackrock   1210.0         0.0017   0.0437   
16  Nuveen Securities   1180.0         0.0025   0.0343   
17          Blackrock   1170.0         0.0020   0.0282   
18  Nuveen Securities   1110.0         0.0030   0.0356   
19               CICC   1060.0         0.0079   0.1732   

                                              Segment   Asset Class  \
0                         Equity: U.S. - Total Market        Equity   
1   Equity: Developed Markets Ex-North America - T...        Equity   
2             Equity: Emerging Markets - Total Market        Equity   
3                     Equity: Global Renewable Energy        Equity   
4                         Equity: U.S. - Total Market        Equity   
5             Equity: Emerging Markets - Total Market        Equity   
6                         Equity: U.S. - Total Market        Equity   
7                         Equity: U.S. - Total Market        Equity   
8                         Equity: U.S. - Total Market        Equity   
9                         Equity: U.S. - Total Market        Equity   
10                    Equity: Global Renewable Energy        Equity   
11                      Equity: U.S. Renewable Energy        Equity   
12              Equity: Global Ex-U.S. - Total Market        Equity   
13                    Equity: Global Renewable Energy        Equity   
14  Fixed Income: U.S. - Broad Market, Broad-based...  Fixed Income   
15                           Equity: U.S. - Small Cap        Equity   
16                     Equity: U.S. - Large Cap Value        Equity   
17                          Equity: Global Low Carbon        Equity   
18                           Equity: U.S. - Small Cap        Equity   
19                 

#### 4) Setting Index:
- Setting the index as `Issuer` and `Ticker` and inner level(1) is Ticker and the outer level(0) is Issuer
- We sort it by issuer by level  0 and then to by the Ticker level 1

In [15]:
#multiindexing the dataframe
df = df.set_index(keys=['Issuer','Ticker']).sort_index(level=[0,1])

In [16]:
#displaying the dataframe
df.head(50)

Fund Name  \
Issuer                       Ticker                                                      
Allianz                      EMNT         PIMCO Enhanced Short Maturity Active ESG ETF   
                             RAFE                              PIMCO RAFI ESG U.S. ETF   
Ameriprise Financial         ESGN    Columbia Sustainable International Equity Inco...   
                             ESGS          Columbia Sustainable U.S. Equity Income ETF   
Barclays                     GRN                             iPath Series B Carbon ETN   
                             RODI                    Barclays Return on Disability ETN   
                             WIL                      Barclays Women in Leadership ETN   
Beyond Investing             VEGN                               U.S. Vegan Climate ETF   
Blackrock                    BGRN                        iShares Global Green Bond ETF   
                             CRBN              iShares MSCI ACWI Low Carbon Target ETF   
                             DMXF                   iShares ESG Advanced MSCI EAFE ETF   
                             DSI                       iShares MSCI KLD 400 Social ETF   
                             EAGG            iShares ESG Aware U.S. Aggregate Bond ETF   
                             EAOA          iShares ESG Aware Aggressive Allocation ETF   
                             EAOK        iShares ESG Aware Conservative Allocation ETF   
                             EAOM            iShares ESG Aware Moderate Allocation ETF   
                             EAOR              iShares ESG Aware Growth Allocation ETF   
                             EMXF                     iShares ESG Advanced MSCI EM ETF   
                             ESGD                      iShares ESG Aware MSCI EAFE ETF   
                             ESGE                        iShares ESG Aware MSCI EM ETF   
                             ESGU                       iShares ESG Aware MSCI USA ETF   
                             ESML             iShares ESG Aware MSCI USA Small-Cap ETF   
                             EUSB       iShares ESG Advanced Total USD Bond Market ETF   
                             HYXF    iShares ESG Advanced High Yield Corporate Bond...   
                             ICLN                      iShares Global Clean Energy ETF   
                             LDEM                      iShares ESG MSCI EM Leaders ETF   
                             SDG                        iShares MSCI Global Impact ETF   
                             SUSA                      iShares MSCI USA ESG Select ETF   
                             SUSB    iShares ESG Aware 1-5 Year USD Corporate Bond ETF   
                             SUSC             iShares ESG Aware USD Corporate Bond ETF   
                             SUSL                     iShares ESG MSCI USA Leaders ETF   
                             USXF                    iShares ESG Advanced MSCI USA ETF   
                             XJH                  iShares ESG Screened S&P Mid-Cap ETF   
                             XJR                iShares ESG Screened S&P Small-Cap ETF   
                             XVV                      iShares ESG Screened S&P 500 ETF   
CICC                         KCCA          KraneShares California Carbon Allowance ETF   
                             KESG         KraneShares MSCI China ESG Leaders Index ETF   
                             KEUA            KraneShares European Carbon Allowance ETF   
                             KGRN    KraneShares MSCI China Clean Technology Index ETF   
                             KRBN                        KraneShares Global Carbon ETF   
Change Finance               CHGX    Change Finance U.S. Large Cap Fossil Fuel Free...   
Compagnie Lombard Odier SCmA ESUS        ETRACS 2x Leveraged MSCI USA ESG Focus TR ETN   
Crestview                    MDCP                    VictoryShares THB Mid Cap ESG ETF   
                           

#### 5) Observation
- Social issues supported by various ETS's are found out by grouping them into different sections
- We can group it by `market` to know the different kinds of social issues like`Global Renewable energy`, `Global Low Carbon` etc

### Analysis

#### 6) Socially responsible ETFs by each issuer
- To get to know the number of ETFs by each issuer, we group the dataframe by issuer. This gives us the details of each issuer 

In [17]:
# creating a group
grouped = df.groupby('Issuer')

- Now to get to know the number of ETFs by each issuer, we loop through each of the observation in the group and print the length of each group which gives us the number of ETFs by each issuer

In [18]:
# looping through the group
for i,group in grouped:
    print(i) , print((len(group)))

Allianz
2
Ameriprise Financial
2
Barclays
3
Beyond Investing
1
Blackrock
27
CICC
5
Change Finance
1
Compagnie Lombard Odier SCmA
1
Crestview
3
Deutsche Bank
10
Direxion
1
EMPIRICAL FINANCE LLC
1
ETFMG
1
Fidelity
1
First Trust
3
Franklin Templeton
4
Goldman Sachs
2
Guinness Atkinson Asset Management
1
Humankind USA LLC
1
Impact Shares
3
Inspire
8
Invesco
9
JPMorgan Chase
1
Mirae Asset Global Investments Co., Ltd.
9
Nationwide
1
New Age Alpha Advisors LLC
1
New York Life
7
Northern Trust
6
Nuveen Securities
12
Point Bridge Capital
1
ProShares
1
ProcureAM
1
Red Gate Advisers LLC
1
Reflection Asset Management, LLC
1
SS&C
1
State Street Global Advisors
8
The Hartford
1
Timothy Plan
2
Toroso Investments
5
Tortoise
1
TrueMark Investments
1
Tuttle Tactical Management, LLC
1
VanEck
3
Vanguard
3
Virtus Investment Partners
1
Wahed
1
WisdomTree
6


In [19]:
#maximum ETF issuer
df.groupby(by='Issuer').size().idxmax()

'Blackrock'

##### Recommendation:
- `Blackrock` is recommended studying further for analysis as it  has the highest number of ETFs issued which is 27

- As, we have multiindexing now by issuer and the ticker, We reset the index for further ease of analysis, to access columns of issuer and ticker and analyze them

In [20]:
#resetting index
df.reset_index()

Issuer Ticker  \
0                 Allianz   EMNT   
1                 Allianz   RAFE   
2    Ameriprise Financial   ESGN   
3    Ameriprise Financial   ESGS   
4                Barclays    GRN   
..                    ...    ...   
161            WisdomTree   IXSE   
162            WisdomTree   RESD   
163            WisdomTree   RESE   
164            WisdomTree   RESP   
165            WisdomTree   XSOE   

                                             Fund Name      AUM  \
0         PIMCO Enhanced Short Maturity Active ESG ETF   177.96   
1                              PIMCO RAFI ESG U.S. ETF    21.37   
2    Columbia Sustainable International Equity Inco...     5.46   
3          Columbia Sustainable U.S. Equity Income ETF     5.61   
4                            iPath Series B Carbon ETN    82.48   
..                                                 ...      ...   
161   WisdomTree India ex-State-Owned Enterprises Fund     7.33   
162                  WisdomTree International ESG Fund     7.78   
163               WisdomTree Emerging Markets ESG Fund    32.65   
164                           WisdomTree U.S. ESG Fund    98.07   
165  WisdomTree Emerging Markets ex-State-Owned Ent...  4280.00   

     Expense Ratio  3-Mo TR  \
0           0.0027  -0.0004   
1           0.0030   0.0360   
2           0.0045  -0.0068   
3           0.0035   0.0468   
4           0.0075   0.1269   
..             ...      ...   
161         0.0058   0.1014   
162         0.0030  -0.0020   
163         0.0032  -0.0250   
164         0.0028   0.0218   
165         0.0032  -0.0013   

                                               Segment   Asset Class  \
0    Fixed Income: U.S. - Broad Market, Broad-based...  Fixed Income   
1                          Equity: U.S. - Total Market        Equity   
2     Equity: Developed Markets Ex-U.S. - Total Market        Equity   
3                          Equity: U.S. - Total Market        Equity   
4                   Commodities: Energy Carbon Credits   Commodities   
..                                                 ...           ...   
161                       Equity: India - Total Market        Equity   
162  Equity: Developed Markets Ex-North America - T...        Equity   
163            Equity: Emerging Markets - Total Market        Equity   
164                        Equity: U.S. - Total Market        Equity   
165            Equity: Emerging Markets - Total Market        Equity   

                                  ETF Segment  \
0    Broad Market, Broad-based Investment Gr…   
1                                Total Market   
2                                Total Market   
3                                Total Market   
4                                        None   
..                                        ...   
161                              Total Market   
162                              Total Market   
163                              Total Market   
164                              Total Market   
165                              Total Market   

                                  Market  
0                                   U.S.  
1                                   U.S.  
2              Developed Markets Ex-U.S.  
3                                   U.S.  
4                  Energy Carbon Credits  
..                                   ...  
161                                India  
162   Developed Markets Ex-North America  
163                     Emerging Markets  
164                                 U.S.  
165                     Emerging Markets  

[166 rows x 10 columns]

#### 7) Total assets under management for each issuer

- Using groupby to group the data by issuer and since we want to know the total number of assets for each issuer, we group the data first by issuer and then add the AUMs of each issuer

In [21]:
# grouping by Issuer and displaying the number of assets under management for each issuer
issuer_aum = df.groupby(by='Issuer')['AUM'].sum()
issuer_aum.sort_values(ascending = False)

Issuer
Blackrock                                   65469.72
Vanguard                                     8783.04
Invesco                                      6268.55
WisdomTree                                   5465.83
Deutsche Bank                                5417.78
Nuveen Securities                            4588.68
First Trust                                  3222.99
Mirae Asset Global Investments Co., Ltd.     1600.00
CICC                                         1296.33
Inspire                                      1128.98
SS&C                                         1010.00
New York Life                                 738.13
Northern Trust                                481.34
Franklin Templeton                            432.51
Goldman Sachs                                 349.68
Allianz                                       199.33
ETFMG                                         191.42
VanEck                                        178.40
Toroso Investments                     

**This is for the recommendation of the Issuer**

- Using groupby to group the data by issuer and since we want to know the mean number of assets for each issuer, we group the data first by issuer and then avergae of the AUMs of each issuer 



In [22]:
issuer_mean_aum = df.groupby(by='Issuer')['AUM'].mean()
issuer_mean_aum.sort_values(ascending = False)

Issuer
Vanguard                                    2927.680000
Blackrock                                   2424.804444
First Trust                                 1074.330000
SS&C                                        1010.000000
WisdomTree                                   910.971667
Invesco                                      895.507143
Deutsche Bank                                541.778000
Nuveen Securities                            382.390000
CICC                                         259.266000
ETFMG                                        191.420000
Mirae Asset Global Investments Co., Ltd.     177.777778
Goldman Sachs                                174.840000
Wahed                                        155.100000
Inspire                                      141.122500
Nationwide                                   119.100000
Franklin Templeton                           108.127500
EMPIRICAL FINANCE LLC                        105.900000
New York Life                            

Recommendation:

- Based on Analysis Since , **Blackrock** has more number of ETFs, it also has the highest AUM in the data.
- Because of the high AUM and multiple ETFs, Blackrock can be recommended to study further in the analysis
- Blackrock is chosen as it has multiple ETFs and more data compared to that of the other issuers, it can be considered as it provides us with more data
- Vanguard has the highest mean AUM, i.e high AUM per ETF, We can also recommend Vanguard as the issuer for further analysis

#### Issuer Recommendations: 
- Blackrock
- Vanguard

#### 8) Total assets under management for each Fund

- Using groupby to group the data by fund name and since we want to know the assets for each issuer, we group the data first by fund name and then sum it because if we put count, the value will be 1 i.e total AUM

In [35]:
# grouping by Fund Name and displaying the assets under management for each fund
fund_aum = df.groupby(by='Fund Name')['AUM'].sum()
fund_aum.sort_values(ascending = False)

Fund Name
iShares ESG Aware MSCI USA ETF                       23430.0
iShares ESG Aware MSCI EAFE ETF                       7110.0
iShares ESG Aware MSCI EM ETF                         7000.0
iShares Global Clean Energy ETF                       6460.0
Vanguard ESG U.S. Stock ETF                           5720.0
                                                      ...   
SPDR S&P Kensho Clean Power ETF                          0.0
SPDR SSGA Gender Diversity Index ETF                     0.0
VictoryShares Top Veteran Employers ETF                  0.0
ETRACS 2x Leveraged MSCI USA ESG Focus TR ETN            0.0
SPDR Bloomberg SASB Corporate Bond ESG Select ETF        0.0
Name: AUM, Length: 166, dtype: float64

- `iShares ESG Aware MSCI USA ETF` is by the issuer `Blackrock`
- Sum of AUM of other multiple ETFs combined are still less than the AUM of one `iShares ESG Aware MSCI USA ETF`.
- `Vanguard ESG U.S. Stock ETF` is by the issuer `Vanguard` and is one of the top 5 ETF that have high AUM
- It shows that it has high capital and can be studies further for analysis

#### Recommended Issuers:
- Vanguard
- Blackrock

#### 9)Expense ratios for the funds of each issuer

- Using groupby to group the data by issuer and since we want to expense ratio for each issuer, we group the data first by issuer.
- We use `mean` expense ratios as it would be better to get to know the expense ratio per ETF to compare it with other issuers
- For Analysis, it would be better to sort the data because we can recommend based on the sorted values, it is given that low expense ratio makes a better investment
- Using `sort_values` we sort the expense ratios

In [24]:
# grouping by issuer and displaying the expenses under each fund
df.groupby(by='Issuer')['Expense Ratio'].mean().sort_values(ascending=True)

Issuer
Humankind USA LLC                           0.001100
Vanguard                                    0.001367
JPMorgan Chase                              0.001500
Deutsche Bank                               0.001690
Blackrock                                   0.001900
Northern Trust                              0.002167
State Street Global Advisors                0.002213
Allianz                                     0.002850
Nuveen Securities                           0.002992
Nationwide                                  0.003400
New York Life                               0.003471
WisdomTree                                  0.003533
Fidelity                                    0.003900
The Hartford                                0.003900
Ameriprise Financial                        0.004000
Tortoise                                    0.004000
VanEck                                      0.004133
Reflection Asset Management, LLC            0.004500
Mirae Asset Global Investments Co., Ltd

- Since we are comparing the different issuers and their expense ratio, It's better if we do it by mean expense ratio as the sum of the expense ratio would not be fair as some issuers have multiple ETFs where some have only 1 

- `HumanKind USA LLC` has the low expense ratio compared to all other issuers but since it has only one investment ETF it is not that preferred for analysis

- Whereas **`Vanguard`** has low expense ratio as well as more AUM next to blackrock and also it's ETF is one of the top 5 AUM for each fund

- **`Vanguard`** and **`Blackrock`** can both be studied for further analysis because of their consistent revies as good investements among the other issuers

#### 10) Frequency distributions of Asset Class, markets and segments for funds of each issuer

##### For frequency of asset classes by issuer:
- We do groupby the issuer and record the asset class values
- For frequency of each segment of the issuer, we use `value_counts` as it gives the count of each asset class in the data
- Since we grouped the data by issuer, it gives the count of each asset class by each issuer

In [25]:
# frequency of segments by asset class
issuer_asset_class = df.groupby(by='Issuer')['Asset Class'].value_counts().reset_index()
issuer_asset_class = issuer_asset_class.set_index(keys=['Issuer','Asset Class']).sort_index(level=[0,1])
issuer_asset_class.rename(columns = {'count':'Frequency'})

Frequency
Issuer                     Asset Class            
Allianz                    Equity                1
                           Fixed Income          1
Ameriprise Financial       Equity                2
Barclays                   Commodities           1
                           Equity                2
...                                            ...
Vanguard                   Equity                2
                           Fixed Income          1
Virtus Investment Partners Equity                1
Wahed                      Equity                1
WisdomTree                 Equity                6

[63 rows x 1 columns]

In [26]:
# other method
issuer_asset = df.groupby(by='Issuer')['Asset Class'].value_counts().reset_index()
issuer_asset

Issuer   Asset Class  count
0                      Allianz        Equity      1
1                      Allianz  Fixed Income      1
2         Ameriprise Financial        Equity      2
3                     Barclays        Equity      2
4                     Barclays   Commodities      1
..                         ...           ...    ...
58                    Vanguard        Equity      2
59                    Vanguard  Fixed Income      1
60  Virtus Investment Partners        Equity      1
61                       Wahed        Equity      1
62                  WisdomTree        Equity      6

[63 rows x 3 columns]

In [27]:
#finding the maximum frequency of segments by issuer
max_issuer_asset = issuer_asset['count'].max()
mask = issuer_asset['count']== max_issuer_asset
issuer_asset[mask]

Issuer Asset Class  count
6  Blackrock      Equity     17

Recommendation:

- The issuer that has the highest AUM in the data is Blackrock and the highest frequency of Asset Class of the issuer is Equity with a count of 17
- Out of the 27 ETFs of Blackrock, 17 are Asset Class of **Equity**
- So factoring on asset class, we can recommend **Equity-Asset Class** can be pursued

##### For frequency of markets by issuer:
- We do groupby the issuer and record the market values
- For frequency of each segment of the issuer, we use `value_counts` as it gives the count of each market in the data
- Since we grouped the data by issuer, it gives the count of each market by each issuer

In [28]:
# frequency of segments by issuer
issuer_market = df.groupby(by='Issuer')['Market'].value_counts().reset_index()
issuer_market

Issuer                               Market  count
0                Allianz                                 U.S.      2
1   Ameriprise Financial                                 U.S.      1
2   Ameriprise Financial            Developed Markets Ex-U.S.      1
3               Barclays                                 U.S.      2
4               Barclays                Energy Carbon Credits      1
..                   ...                                  ...    ...
95            WisdomTree                     Emerging Markets      2
96            WisdomTree   Developed Markets Ex-North America      1
97            WisdomTree                                India      1
98            WisdomTree                                China      1
99            WisdomTree                                 U.S.      1

[100 rows x 3 columns]

In [29]:
#finding the maximum frequency of markets by issuer
max_issuer_market = issuer_market['count'].max()
mask2 = issuer_market['count'] == max_issuer_market
issuer_market[mask2]

Issuer Market  count
6  Blackrock   U.S.     14

Recommendation:

- The issuer that has the highest AUM in the data is Blackrock and the highest frequency of the issuer is US Market with a count of 14
- Out of the 27 ETFs of Blackrock, 14 has their market in the **US**
- So factoring on market, we can recommend **Market-US** can be pursued

##### For frequency of segments by issuer:
- We do groupby the issuer and record the segment values
- For frequency of each segment of the issuer, we use `value_counts` as it gives the count of each segment in the data
- Since we grouped the data by issuer, it gives the count of each segment by each issuer

In [30]:
# frequency of segments by issuer
issuer_segment = df.groupby(by='Issuer')['ETF Segment'].value_counts().reset_index()
issuer_segment

Issuer                               ETF Segment  count
0                Allianz  Broad Market, Broad-based Investment Gr…      1
1                Allianz                              Total Market      1
2   Ameriprise Financial                              Total Market      2
3               Barclays                              Total Market      1
4               Barclays                                 Large Cap      1
..                   ...                                       ...    ...
80                VanEck                 Broad Market, Broad-based      1
81              Vanguard                              Total Market      2
82              Vanguard   Corporate, Broad-based Investment Grade      1
83                 Wahed                              Total Market      1
84            WisdomTree                              Total Market      6

[85 rows x 3 columns]

In [31]:
#finding the maximum frequency of segments by issuer
max_issuer_segment = issuer_segment['count'].max()
mask = issuer_segment['count']== max_issuer_segment
issuer_segment[mask]

Issuer   ETF Segment  count
6  Blackrock  Total Market     11

Recommendation:

- The issuer that has the highest AUM in the data is Blackrock and the highest frequency of ETF-Segment of the issuer is Total Market with a count of 11
- Out of the 27 ETFs of Blackrock, 11 has their segment in the **Total Market**
- So factoring on segment, we can recommend **segment-Total** can be pursued

#### 11) 3 Month Total Return Percentages:
- Since we need a combination of asset class, market and segment, we group them together by asset class, market and segment and reset the index, which gives different combinations of asset class, market and segment

In [32]:
#3 month total return percentages factoring in combination of asset class, market and segment together
return_mean = df.groupby(['Asset Class', 'Market', 'ETF Segment'])['3-Mo TR'].mean().reset_index()
return_mean.sort_values(by = '3-Mo TR', ascending = False)

Asset Class                               Market  \
12            Equity                                India   
21            Equity                                 U.S.   
13            Equity                        North America   
17            Equity                                 U.S.   
19            Equity                                 U.S.   
16            Equity                                 U.S.   
14            Equity                                 U.S.   
22            Equity                                 U.S.   
7             Equity                               Global   
20            Equity                                 U.S.   
6             Equity                               Global   
23            Equity                                 U.S.   
18            Equity                                 U.S.   
0             Equity                                China   
8             Equity                               Global   
1             Equity                    Developed Markets   
10            Equity                       Global Ex-U.S.   
3             Equity            Developed Markets Ex-U.S.   
9             Equity                               Global   
2             Equity   Developed Markets Ex-North America   
11            Equity                       Global Ex-U.S.   
5             Equity                     Emerging Markets   
30      Fixed Income                                 U.S.   
24      Fixed Income                    Developed Markets   
4             Equity            Developed Markets Ex-U.S.   
29      Fixed Income                                 U.S.   
32      Fixed Income                                 U.S.   
28      Fixed Income                                 U.S.   
27      Fixed Income                               Global   
31      Fixed Income                                 U.S.   
26      Fixed Income                               Global   
25      Fixed Income                     Emerging Markets   
15            Equity                                 U.S.   
33  Leveraged Equity                                 U.S.   

                                 ETF Segment   3-Mo TR  
12                              Total Market  0.101400  
21                             Mid Cap Value  0.061700  
13                              Total Market  0.058500  
17                          Large Cap Growth  0.054300  
19                                   Mid Cap  0.050600  
16                                 Large Cap  0.048776  
14                           Extended Market  0.044800  
22                                 Small Cap  0.041200  
7                           Large Cap Growth  0.039300  
20                            Mid Cap Growth  0.034100  
6                                  Large Cap  0.033000  
23                              Total Market  0.032509  
18                           Large Cap Value  0.030600  
0                               Total Market  0.029300  
8                               Total Market  0.025867  
1                               Total Market  0.024950  
10                                 Large Cap  0.023400  
3                                  Large Cap  0.021850  
9                        Total Market Growth  0.014200  
2                               Total Market  0.013667  
11                              Total Market  0.012700  
5                               Total Market  0.009822  
30         Corporate, Broad-based High Yield  0.000200  
24               Corporate, Broad-based Hig… -0.001400  
4                               Total Market -0.006800  
29  Broad Market, Broad-based Investment Gr… -0.008133  
32  Corporate, Broad-based Investment Grade… -0.010550  
28                 Broad Market, Broad-based -0.010700  
27    Broad Market, Broad-based Investment … -0.011050  
31   Corporate, Broad-based Investment Grade -0.012750  
26                 Broad Market, Broad-based -0.016500  
25              Government, Non-Native Curr… -0.0

- In the previous recommendations that we made about class and segment, we recommended asset class and total market as their frequency is high.
- From this analysis, it is recommended to have **Equity** as the Asset Class and **Total Market** as the segment as it's mean 3-Month Total revenue percentage is high and it indicates a good investment
- **India** can be selected as market as it had the highest 3-Month Total Revenue Percentage
- The frequency of high 3-Month Total Revenue percentage of the US market is high So **US** market can also be recommended.

In [33]:
# finding the mean 3- Month Total return percentage by issuer
df.groupby(by='Issuer')['AUM'].mean().sort_values(ascending=False)

Issuer
Vanguard                                    2927.680000
Blackrock                                   2424.804444
First Trust                                 1074.330000
SS&C                                        1010.000000
WisdomTree                                   910.971667
Invesco                                      895.507143
Deutsche Bank                                541.778000
Nuveen Securities                            382.390000
CICC                                         259.266000
ETFMG                                        191.420000
Mirae Asset Global Investments Co., Ltd.     177.777778
Goldman Sachs                                174.840000
Wahed                                        155.100000
Inspire                                      141.122500
Nationwide                                   119.100000
Franklin Templeton                           108.127500
EMPIRICAL FINANCE LLC                        105.900000
New York Life                            

In [34]:
df.groupby(by='Issuer')['Expense Ratio'].mean().sort_values(ascending=True)

Issuer
Humankind USA LLC                           0.001100
Vanguard                                    0.001367
JPMorgan Chase                              0.001500
Deutsche Bank                               0.001690
Blackrock                                   0.001900
Northern Trust                              0.002167
State Street Global Advisors                0.002213
Allianz                                     0.002850
Nuveen Securities                           0.002992
Nationwide                                  0.003400
New York Life                               0.003471
WisdomTree                                  0.003533
Fidelity                                    0.003900
The Hartford                                0.003900
Ameriprise Financial                        0.004000
Tortoise                                    0.004000
VanEck                                      0.004133
Reflection Asset Management, LLC            0.004500
Mirae Asset Global Investments Co., Ltd

#### 12) Socially Responsible ETFs vs Non Socially Responsible ETFs

-  Direxion Daily S&P 500® Bull and Bear 3X Share has 2 ETFs leveraged and inverse, which are designed to provide investors with amplified exposure to the daily performance of the S&P 500 Index.
- These are not socially responsible ETFs but according to the data in the reference, Their expense ratio is 1% and 1.08%
- The expense ratio for the Socially responsible ETFs has a mean of less than 1%
- Since having a low expense ratio relates for a better investment, Socially responsible ETFs are a better investment and are more profitable

-- https://www.direxion.com/product/daily-sp-500-bull-bear-3x-etfs?keyword=large%20cap%20etfs&gad=1&gclid=Cj0KCQjwtJKqBhCaARIsAN_yS_lCMevB8eve2ciQDUCMfJCkc0jwWRCM3ARB0xt9Xjrh5eNwK-Fk_6UaApXjEALw_wcB

One more Reference:

- These are the general higehst monthly returns which have an some of the ETFs whose AUM is less than a million
- Whereas in our socially responsible ETFs data, barely no ETF have less than a million
- This also proves that Socially responsible ETF is more profitable than those ETFs which have no social responsibility

https://etfdb.com/compare/highest-monthly-returns/

"A meta-analysis by the NYU Stern Center for Sustainable Business of more than 1,000 research papers published between 2015 and 2020 found that among studies focused on risk-adjusted attributes, 59% found that sustainable options performed as well or better than conventional approaches while only 14% saw a negative result."

- This point also says that the socially responsible ETFs are more profitable than the ETFs which have no social responsibility.

- #https://www.forbes.com/advisor/investing/sri-socially-responsible-investing/


"I pledge on my honor that I have not given nor received any unauthorized assistance on this assignment."
-Anusha Aligeti